src/sentry/api/serializers/models/project.py

```diff
@@ -93,22 +93,44 @@ def get_access_by_project(
     for pt in project_teams:
         project_team_map[pt.project_id].append(pt.team)
 
-    team_memberships = set(_get_team_memberships([pt.team for pt in project_teams], user))
-    all_org_roles = get_org_roles({i.organization_id for i in projects}, user)
-    prefetch_related_objects(projects, "organization")
+    team_memberships = _get_team_memberships([pt.team for pt in project_teams], user)
 
+    org_ids = {i.organization_id for i in projects}
+    all_org_roles = get_org_roles(org_ids, user)
     is_superuser = request and is_active_superuser(request) and request.user == user
+    prefetch_related_objects(projects, "organization")
+
     result = {}
     for project in projects:
-        is_member = any(t.id in team_memberships for t in project_team_map.get(project.id, []))
+        parent_teams = [t.id for t in project_team_map.get(project.id, [])]
```

Comment:
what's a parent team?

can we have teams within teams??

Reply:
A project can be a child of several teams.